src/core/io/platform/f3xx/PWM_INPUTf3xx.cpp

```diff
@@ -0,0 +1,289 @@
+
+#include <core/io/platform/f3xx/GPIOf3xx.hpp>
+#include <core/io/platform/f3xx/PWM_INPUTf3xx.hpp>
+
+namespace core::io {
+
+// Global pointer to the active instance (one at a time)
+static PWM_INPUTf3xx* activePwmInput = nullptr;
+
+/**
+ * Get timer instance, direct channel, indirect channel, alternate function,
+ * triggerSource, irqNumber, and activeChannel associated with a pin.
+ * This information is pulled from the STM32F302x8 documentation.
+ *
+ * NOTE: PA_2 and PA_3 are used for UART Tx and Rx respectively
+ *
+  @param pin The pin to check the instance of
+* @param instance The timer instance to assign to
+* @param directChannel The direct channel to assign to
+* @param indirectChannel The indirect channel to assign to
+* @param alternateFunction The GPIO identifier for the function of the pin
+* @param irqNumber IRQ number for NVIC
+* @param activeChannel HAL enum for which channel will fire the interrupt
+ */
+
+static void getInputInstance(Pin pin, TIM_TypeDef** instance, uint32_t* directChannel, uint32_t* indirectChannel,
+                             uint32_t* alternateFunction, uint32_t* triggerSource, IRQn_Type* irqNumber,
+                             uint32_t* activeChannel) {
+    switch (pin) {
+#if defined(STM32F302x8)
+    // TIM 1 Channel 1 Direct, Channel 2 Indirect
+    case Pin::PC_0:
+        *instance          = TIM1;
+        *directChannel     = TIM_CHANNEL_1;
+        *indirectChannel   = TIM_CHANNEL_2;
```

Comment:
This will need to change with my timer updates. I will handle this, once PWM input get's merged.